packages/firestore/test/unit/specs/spec_builder.ts

```diff
@@ -936,14 +933,55 @@ export class SpecBuilder {
     }
   }
 
+  private addQueryToActiveTargetsIfNotYet(
+    targetId: number,
+    query: Query,
+    resumeToken?: string
+  ): void {
+    if (this.activeTargets[targetId]) {
+      const specQueriesClone = [...this.activeTargets[targetId].queries];
+      // query is not added to active targets yet.
+      if (
+        !specQueriesClone.some(specQuery =>
+          parseQuery(specQuery).isEqual(query)
+        )
+      ) {
+        specQueriesClone.push(SpecBuilder.queryToSpec(query));
```

Comment:
This would flow a bit better if line 942, 943, and 951 onwards were inside of the if statement. You could even inline specQueriesClone.

Reply:
Done.

I still keep the temp variable to save some typing.